tarok/src/cards.cpp

```diff
@@ -84,4 +89,31 @@ std::array<Card, 54> BuildDeck() {
   return deck;
 }
 
+std::tuple<std::vector<int>, std::array<std::vector<int>, 3>> DealCards(
+    int seed = 42) {
+  // create vector of card indices
+  std::vector<int> cards(54);
+  std::iota(std::begin(cards), std::end(cards), 0);
+
+  // shuffle indices
+  auto rng = std::default_random_engine(seed);
+  std::shuffle(std::begin(cards), std::end(cards), rng);
+
+  // first six cards are talon
+  std::vector<int>::iterator start(cards.begin());
```

Comment:
`std::vector<int>::iterator start = cards.begin();` looks better to me but this is a personal opinion :).